content/news/2019-12-03-mapstruct-and-quarkus.md

```diff
@@ -0,0 +1,220 @@
+This allows having the advantages of native applications also for Java-based apps like a much faster startup time, less memory usage and a much smaller image (just a few megabytes instead of easily more than 100 MB for the JVM and required libraries).
+
+Using a native image is not possible for all kinds of applications, especially when reflection is used the GraalVM compiler needs some assistance by the developer in order to create the native image. That means you have to provide reflection metadata to the compiler which makes the development more complicated.
+
+And thats exactly the point where MapStruct jumps in as a reflection-free mapping library!
+
+### How to integrate MapStruct in a Quarkus application
+So let us see how well MapStruct plays together with Quarkus!
+
+I made an example application that will show you how you can integrate MapStruct in your Quarkus application. You will find the complete demo application in our [examples repository](https://github.com/mapstruct/mapstruct-examples/mapstruct-quarkus).
+
+The example should be a simple REST application just having a `/person` endpoint returning information about a person. Obviously, the internally used classes should not be exposed, but a DTO will be created and filled with information from a MapStruct mapper.
+
+#### Create a new Quarkus application
+After reading the [Getting Started](https://quarkus.io/guides/getting-started-guide) on the Quarkus pages it was really easy to figure out how to create a new application. Just run the following command:
+
+{{< prettify >}}
+mvn io.quarkus:quarkus-maven-plugin:1.0.1.Final:create \
+  -DprojectGroupId=org.mapstruct.examples.quarkus \
+  -DprojectArtifactId=mapstruct-examples-quarkus \
+  -DclassName="org.mapstruct.example.quarkus.PersonResource" \
+  -Dpath="/person" \
+  -Dextensions="resteasy-jsonb"
+{{< /prettify >}}
+
+This will setup a new Quarkus application having a REST endpoint `/person` defined in a `PersonResource`. We want to expose the JSON structure automatically, so we need the `resteasy-jsonb` extension that will allow us to marshall our object to JSON (like you probably know it from Spring REST or similar frameworks).
+
+#### Adding MapStruct
+You just need to add MapStruct in your projects POM:
+
+{{< prettify xml >}}
+<dependency>
+  <groupId>org.mapstruct</groupId>
+  <artifactId>mapstruct</artifactId>
+  <version>1.3.1.Final</version>
+</dependency>
+<dependency>
+  <groupId>org.mapstruct</groupId>
+  <artifactId>mapstruct-processor</artifactId>
+  <version>1.3.1.Final</version>
+  <scope>provided</scope>
+</dependency>
+{{< /prettify >}}
+
+_Side note:_ In case you use for example Lombok pay attention that you define the dependency **before** the one for the MapStruct processor.
```

Comment:
```suggestion
_Side note:_ In case you use for example Lombok pay attention that you must define the Lombok dependency **before** the one for the MapStruct processor.
```